File: java/chainofresponsibility/AuthorizationMiddlewareHandler.java
package chainofresponsibility;

public class AuthorizationMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public boolean handle(Request request) {
        System.out.println("Checking if request is authorized...");
        if (!request.isAuthorized()) {
            System.out.println("request failed to be authorized...");
            return false;
        }
        return super.handle(request);
    }
}

File: typescript/chainofresponsibility/authorization-middleware-handler.ts
import {AbstractMiddlewareHandler} from "./abstractMiddleware-handler";
import {ChainRequest} from "./request";

export class AuthorizationMiddlewareHandler extends AbstractMiddlewareHandler {
  public handle(request: ChainRequest): boolean {
    if (!request.isAuthorized) {
      console.log("AuthorizationMiddlewareHandler: User is not authorized.");
      return false;
    }
    console.log("AuthorizationMiddlewareHandler: User is authorized.");

    return super.handle(request);
  }
}
